File: Level1/Materal.BusinessFlow/Src/Materal.BusinessFlow.Abstractions/Domain/Node.cs
using System.ComponentModel.DataAnnotations;

namespace Materal.BusinessFlow.Abstractions.Domain
{
    /// <summary>
    /// 节点
    /// </summary>
    public class Node : BaseDomain
    {
        /// <summary>
        /// 名称
        /// </summary>
        [Required, StringLength(40)]
        public string Name { get; set; } = string.Empty;
        /// <summary>
        /// 步骤唯一标识
        /// </summary>
        [Required]
        public Guid StepID { get; set; }
        /// <summary>
        /// 处理类型
        /// </summary>
        [Required]
        public NodeHandleTypeEnum HandleType { get; set; }
        /// <summary>
        /// 执行条件
        /// </summary>
        public string? RunConditionExpression { get; set; }
        /// <summary>
        /// 数据
        /// </summary>
        public string? Data { get; set; }
        /// <summary>
        /// 处理数据
        /// </summary>
        public string? HandleData { get; set; }
    }
}

File: Level3/Materal.BusinessFlow/Demo/BusinessFlowWebDemo/src/models/Node/Node.ts
import { BaseDomain } from "../BaseDomain";
import { NodeHandleTypeEnum } from "./NodeHandleTypeEnum";

export class Node extends BaseDomain {
    /**
     * 名称
     */
    public Name: string = "";
    /**
     * 步骤唯一标识
     */
    public StepID: string = "";
    /**
     * 处理类型
     */
    public HandleType: NodeHandleTypeEnum = NodeHandleTypeEnum.Auto;
    /**
     * 执行条件
     */
    public RunConditionExpression?: string;
    /**
     * 数据
     */
    public Data?: string;
    /**
     * 处理数据
     */
    public HandleData?: string;
}